packages/local_auth/local_auth_darwin/darwin/local_auth_darwin/Sources/local_auth_darwin/LocalAuthPlugin.swift

```diff
@@ -0,0 +1,254 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Foundation
+import LocalAuthentication
+
+#if canImport(FlutterMacOS)
+  import FlutterMacOS
+#elseif canImport(Flutter)
+  import Flutter
+#endif
+
+class StickyAuthState {
+  let options: AuthOptions
+  let strings: AuthStrings
+  let resultHandler: (Result<AuthResultDetails, Error>) -> Void
+
+  init(
+    options: AuthOptions, strings: AuthStrings,
+    resultHandler: @escaping (Result<AuthResultDetails, Error>) -> Void
+  ) {
+    self.options = options
+    self.strings = strings
+    self.resultHandler = resultHandler
+  }
+}
+
+public final class LocalAuthPlugin: NSObject, FlutterPlugin, LocalAuthApi {
+  private var authContext: AuthContext
```

Comment:
Why have you switched to a singe persistent auth context? `AuthContext` is stateful, so re-using it from call to call could have unintended consequences. Things should only be persistent when there's a reason we want to preserve state.

Reply:
You are right in saying that. I made this change with the aim of reuse, but I forgot that if any "contamination" occurs, this can lead to subsequent errors.

Reply:
For `AuthContext`, I implemented a closure-based provider pattern that offers a different but effective approach compared to the protocol + implementation used for AlertController.

Reply:
> I implemented a closure-based provider pattern that offers a different but effective approach

Why have you changed the approach? Was there an issue using the previous approach, in order to minimize unrelated changes?